packages/app/src/specs/SpecsListCloudButton.vue

```diff
@@ -0,0 +1,141 @@
+<template>
+  <Button
+    v-if="!!buttonOptions.text"
+    data-cy="cloud-button"
+    variant="linkBold"
+    :prefix-icon="buttonOptions.icon"
+    :disabled="!buttonOptions.emits"
+    prefix-icon-class="icon-light-indigo-100 icon-dark-indigo-500"
+    @click.stop.prevent="handleClick"
+  >
+    <span class="hidden lg:inline">{{ buttonOptions.text }}</span>
+    <span class="lg:hidden">{{ buttonOptions.textShort }}</span>
+  </Button>
+</template>
+
+<script setup lang="ts">
+import Button from '@cy/components/Button.vue'
+import type { FunctionalComponent, SVGAttributes } from 'vue'
+import UserIcon from '~icons/cy/user-outline_x16.svg'
+import ChainIcon from '~icons/cy/chain-link_x16.svg'
+import PlayIcon from '~icons/cy/play-large_x16.svg'
+import { computed } from 'vue'
+import { gql } from '@urql/vue'
+import type { SpecCloudDataHoverButtonFragment } from '../generated/graphql'
+import { useI18n } from '@cy/i18n'
+
+const { t } = useI18n()
+
+const emits = defineEmits<{
+  (eventName: 'showLogin'): void
+  (eventName: 'showConnectToProject'): void
+  (eventName: 'requestAccess'): void
+}>()
+
+const props = defineProps<{
+  gql: SpecCloudDataHoverButtonFragment
+}>()
+
+gql`
+fragment SpecCloudDataHoverButton on Query {
+  currentProject {
+    id
+    cloudProject{
+      __typename
+      ... on CloudProjectUnauthorized {
+        hasRequestedAccess
+      }
+    }
+  }
+  ...Auth
+  ...CloudConnectModals
+  ...RequestAccessButton
+}
+`
+
+const projectConnectionStatus = computed(() => {
```

Comment:
Nice use of `computed` here.

Reply:
Also stole this pattern, but it is nice for sure.